src/Utilities/MuxLogger.cs

```diff
@@ -1341,7 +1346,11 @@ private void UnregisterAllEventHandlers()
                 _eventSourceForBuild.TaskFinished -= _taskFinishedEventHandler;
                 _eventSourceForBuild.TaskStarted -= _taskStartedEventHandler;
                 _eventSourceForBuild.WarningRaised -= _buildWarningEventHandler;
-                _eventSourceForBuild.TelemetryLogged -= _telemetryEventHandler;
+
+                if (_eventSourceForBuild is IEventSource2)
+                {
+                    ((IEventSource2)_eventSourceForBuild).TelemetryLogged -= _telemetryEventHandler;
```

Comment:
I think the usual advice is to do `as` and check for `null`, instead of checking for `is` followed by a cast.